File: exercise/scala/day08/src/main/scala/service/ToyProductionService.scala
package service

import domain.Toy.State.*
import domain.{Toy, ToyRepository}

class ToyProductionService(private val repository: ToyRepository) {
  def assignToyToElf(toyName: String): Unit = {
    val toyOpt = repository.findByName(toyName)
    toyOpt.foreach { toy =>
      if (toy.state == UNASSIGNED) {
        toy.state = IN_PRODUCTION
        repository.save(toy)
      }
    }
  }
}

File: solution/day08/src/main/java/service/ToyProductionService.java
package service;

import domain.ToyRepository;

public class ToyProductionService {
    private final ToyRepository repository;

    public ToyProductionService(ToyRepository repository) {
        this.repository = repository;
    }

    public void assignToyToElf(String toyName) {
        repository.findByName(toyName)
                .ifPresent(toy -> {
                    toy.assignToElf();
                    repository.save(toy);
                });
    }
}